In [9]:
import re

# Logging line example : [ 67998][I][motorControl.cpp:97] run(): [MotorControl] Current: 418.549988
# Use a regular expression to extract the current value
# The regular expression is: \d+\.\d+

#!/usr/bin/env python3

import re
import pandas as pd
import os

def parsemyFile(filename):
    # Open the file for reading and read all the lines.
    # Parse each line using the parsemystring function and put the results in a list
    # Then plot the list using matplotlib as an XY-plot

    results = []

    with open(filename, 'r') as f:
        lines = f.readlines()
        for line in lines:
            timestamp, current = parsemystring(line)
            if not timestamp is None and not current is None:
                results.append([timestamp, current])
    return results


def parsemystring(mystring):
    # Use a regular expression to extract the timestamp and the current value
    # example line : "[179726][I][motorControl.cpp:97] run(): [MotorControl] Current: 1493.400024"
    # The regular expression is: \[(\d+)\].*Current: (\d+\.\d+)

    m = re.match(r'\[\W*(\d+)\].*Current: (\d+\.\d+)', mystring)
    if not m is None:
        timestamp = m.group(1)
        current = m.group(2)
        return timestamp, current
    else:
        return None, None

# result = parsemyFile('4.5V_lettingDown.log')
# result = parsemyFile('4.5V_pullingUp.log')
# result = parsemyFile('6V_lettingdown.log')
result = parsemyFile('6V_pullingUp.log')

data = pd.DataFrame(result, columns=['timestamp [ms]', 'current [mA]'])

#all columns are strings, convert to numeric otherwise NaN
data = data.apply(pd.to_numeric, errors='coerce')
data['timestamp [ms]'] = data['timestamp [ms]'] - data['timestamp [ms]'].min()

print("got % d lines" % len(data))
data.head()

print('Average current: ', round(data['current [mA]'].mean(),1), 'mA')
print('Max current: ', round(data['current [mA]'].max(),1), 'mA')
print('Min current: ', round(data['current [mA]'].min(),1), 'mA')
print('Standard deviation: ', round(data['current [mA]'].std(),2), 'mA \n')

got  235 lines
Average current:  1635.4 mA
Max current:  1770.6 mA
Min current:  1420.0 mA
Standard deviation:  66.26 mA 



# VMOTOR = 4.5V
## Lowering the door
Average current:  434.1 mA
Max current:  539.0 mA
Min current:  383.5 mA
Standard deviation:  23.76 mA 

## Raising the door
Average current:  1363.2 mA
Max current:  1523.9 mA
Min current:  1201.5 mA
Standard deviation:  87.81 mA 

# VMOTOR = 6.0V
## Lowering the door
Average current:  522.9 mA
Max current:  581.3 mA
Min current:  456.0 mA
Standard deviation:  30.42 mA 

## Raising the door
Average current:  1635.4 mA
Max current:  1770.6 mA
Min current:  1420.0 mA
Standard deviation:  66.26 mA 

In [12]:
# Lowering door
I_45 = 434.1
I_6 = 522.9

ratio_lowering = I_6 / I_45
print('Ratio lowering: ', round(ratio_lowering,2))

# Raising door
I_45 = 1363.2
I_6 = 1635.4
ratio_lifting = I_6 / I_45
print('Ratio lifting: ', round(ratio_lifting,2))


Ratio lowering:  1.2
Ratio lifting:  1.2
